### ISSUE TITLE
SystemExit and sys.exit behavior overlap

### ISSUE THREAD

**Issue by thecnoNSMB** [wemake-services/wemake-python-styleguide#1786]
# Rule request

`raise SystemExit(arg)` and `sys.exit(arg)` are identical in behavior, one should be favored over the other for consistency.

## Reasoning

I don't know which of the two would be preferable for this linter (I suspect `sys.exit(arg)`, but `raise SystemExit(arg)` doesn't require any imports), but several other contexts where multiple forms are valid have a rule enforcing one form as the standard (such as single vs double quotes, or string formatting methods), and I figure the same logic should apply here.

**Comment by thecnoNSMB** [comment]
`quit(arg)` and `exit(arg)` should also be prohibited under this rule, as those are from the `site` module and are intended to be used for the REPL; as such, they may not be loaded at all when a script runs.

**Comment by orsinium** [comment]
> I suspect sys.exit(arg), but raise SystemExit(arg) doesn't require any imports

Yes, I am for `sys.exit`.

1. Importing of `sys` is incredibly fast, I'd even say free:

    ```python
    import sys
    sys.__loader__  # _frozen_importlib.BuiltinImporter
    sys.meta_path.index(sys.__loader__)  # 0
    ```

2. It should be clear that this statement isn't an exception that can be caught somewhere up in the stack but an uncompromised exit without turning back.

> quit(arg) and exit(arg) should also be prohibited under this rule

WPS already has this one:

```
WPS421 | Found wrong function call: exit
```

However, I guess it should be more verbose like in pylint:

```
R1722 | Consider using sys.exit()
```

**Comment by thecnoNSMB** [comment]
It doesn't look like your commit is correct, or at least it doesn't handle this issue. `SystemError` is raised on internal interpreter bugs, and probably shouldn't be raised by the programmer in any case, but we were talking about `SystemExit`, a different exception explicitly intended to cleanly exit the program.

**Comment by sobolevn** [comment]
OMG, thank you! I was in a total hurry and made a typo.
